kyo-combinators/shared/src/main/scala/kyo/KYO.scala

```diff
@@ -0,0 +1,209 @@
+package kyo
+
+import kyo.Envs.HasEnvs
+import scala.annotation.implicitNotFound
+import scala.annotation.tailrec
+import scala.annotation.targetName
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+import scala.util.NotGiven
+
+
+object KYO:
```

Comment:
Would prefer `Kyo`.

Reply:
I had made it KYO after ZIO, but I'm not sure we need to keep it that way. Is there a `Kyo` type anywhere this would clash with?

Reply:
There isn't a user-facing Kyo type yet but I'd prefer to keep it reserved for kyo-core. In a sense, the pending type is an expression of the Kyo monad :) 

We don't have an IO type yet. Perhaps we could define one in kyo-core and then provide extensions for both the class and companion object in kyo-combinators?

Something that would be interesting to try is defining IO as a case class. Considering the target user base of these extensions, allowing arbitrary nesting of computations better matches the mental model of how this kind of API is traditionally used. There should be a performance cost but it might not be significant since the core might still be inlinable by the JIT compiler and this kind of temporary wrapper is easy to virtualize.

I'm thinking something like:

case class IO[T, S](kyo: T < S)

object IO:
    // not sure if we need methods here, maybe only extensions in kyo-combinators?

// keep IOs as the effect (I know we need to talk about plurals 😅)
object IOs:
    def get[T, S](io: IO[T, S]) = ...
    // current IOs methods

Reply:
Actually, I think `IO` wouldn't make sense unless we want to have the `IOs` effect by default. Maybe it could be `Kyo` but in `kyo-core` with extensions in `kyo-combinators`. We could still provide a `map` for it but taking a function `T => Kyo[U, S2] | U < S2`.

Reply:
Agreed that `IO` isn't the ideal object for this method.

I just added a `Kyo` object to the `kyo` package in `kyo-core` and turned the `KYO` object definition into extension methods on it.

Reply:
thanks! I'm exploring [another way](https://github.com/getkyo/kyo/issues/462) to support nesting for any computation so I think we can go without a `case class Kyo` for now.